src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let copy = { ...state };
+  const res = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copy, action.extraData);
+        break;
+
+      case 'removeProperties' :
+        for (const prop of action.keysToRemove) {
+          delete copy[prop];
+        }
 
+        break;
+
+      case 'clear' :
+        copy = {};
+        break;
```

Comment:
It is a good practice to add `default` case for `switch` statement
If you don't have a default behavior - just throw an Error with text "incorrect action type"